jsonnet/lib/observatorium-api.libsonnet

```diff
@@ -101,51 +113,79 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
       container.mixin.readinessProbe.httpGet.withPort(api.config.ports.internal) +
       container.mixin.readinessProbe.httpGet.withScheme('HTTP') +
       container.mixin.readinessProbe.httpGet.withPath('/ready') +
-      if api.config.rbac != {} || api.config.tenants != {} then
-        container.withVolumeMounts(
-          (if api.config.rbac != {} then [
-             {
-               name: 'rbac',
-               mountPath: '/etc/observatorium/rbac.yaml',
-               subPath: 'rbac.yaml',
-               readOnly: true,
-             },
-           ] else []) +
-          (if api.config.tenants != {} then [
-             {
-               name: 'tenants',
-               mountPath: '/etc/observatorium/tenants.yaml',
-               subPath: 'tenants.yaml',
-               readOnly: true,
-             },
-           ] else [])
-        ) else {};
-
-    deployment.new(api.config.name, api.config.replicas, c, api.config.commonLabels) +
-    deployment.mixin.metadata.withNamespace(api.config.namespace) +
-    deployment.mixin.metadata.withLabels(api.config.commonLabels) +
-    deployment.mixin.spec.selector.withMatchLabels(api.config.podLabelSelector) +
-    deployment.mixin.spec.strategy.rollingUpdate.withMaxSurge(0) +
-    deployment.mixin.spec.strategy.rollingUpdate.withMaxUnavailable(1) +
-    if api.config.rbac != {} || api.config.tenants != {} then
-      deployment.mixin.spec.template.spec.withVolumes(
+      container.withVolumeMounts(
         (if api.config.rbac != {} then [
            {
-             configMap: {
-               name: api.config.name,
-             },
              name: 'rbac',
+             mountPath: '/etc/observatorium/rbac.yaml',
+             subPath: 'rbac.yaml',
+             readOnly: true,
            },
          ] else []) +
         (if api.config.tenants != {} then [
            {
-             secret: {
-               secretName: api.config.name,
-             },
              name: 'tenants',
+             mountPath: '/etc/observatorium/tenants.yaml',
+             subPath: 'tenants.yaml',
+             readOnly: true,
+           },
+         ] else []) +
+        (if api.config.tls != {} then [
+           {
+             name: 'observatorium-api-tls-secret',
```

Comment:
Nit: let's try to keep resource names stutter-free, i.e. we don't need to append `secret` to the madness of our secrets, just like we don't add `deployment` to the madness of our deployments

Reply:
I still need to separate the configMap and secret mounts. how should I keep the names seperate?

Reply:
so I cannot name  both secret and configmap `observatorium-api-tls, (will override each other in volume mounts. how should those be named?

Reply:
hmm sorry I really mean this in the volumes section, not the volume mounts. The secret's own name should not contain the word `secret`. It is clear it is a secret already :)